File: python/0007.reverse-integer/reverse-integer.java
public class Solution {
    public int reverse(int x) {
        boolean negtiveFlag = false;
        if (x < 0) {
            negtiveFlag = true;
            // reverse to postive number;
            x = -x;
        }

        char[] valueOfOrigin = String.valueOf(x).toCharArray();

        int len = valueOfOrigin.length;
        for (int i = 0; i < len / 2; i++) {
            // swap reverse order
            char temp = valueOfOrigin[i];
            valueOfOrigin[i] = valueOfOrigin[len - 1 - i];
            valueOfOrigin[len - 1 - i] = temp;
        }
        String positiveReversed = new String(valueOfOrigin);
        int result = 0;
        try {
            result = Integer.parseInt(negtiveFlag ? "-" + positiveReversed : positiveReversed);
        } catch (NumberFormatException ex) {
            result = 0;
        }
        return result;
    }
}

File: python/0007.reverse-integer/reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        # 核心思想还是用一个大于32位容器去装翻转数字
        sign = -1 if  x < 0 else 1
        x = abs(x)
        ans = 0
        while x:
            ans = ans * 10 + x % 10
            x //= 10
        return sign * ans if ans <= 0x7fffffff else 0
